lib/fw.c

```diff
@@ -474,23 +500,24 @@ int switchtec_fw_part_info(struct switchtec_dev *dev, int nr_info,
 		if (ret)
 			return ret;
 
-
 		if (info[i].type == SWITCHTEC_FW_TYPE_NVLOG) {
 			inf->version[0] = 0;
 			inf->crc = 0;
 			continue;
 		}
 
-		ret = switchtec_fw_read_footer(dev, inf->image_addr,
-					       inf->image_len, &ftr,
-					       inf->version,
-					       sizeof(inf->version));
+		if (ret)
```

Comment:
please review the code more carefully, this is very low level code defect.

Reply:
yes, it should be 'if (ret < 0)', fix it.